### PR TITLE
Add iptables manager for tunnel port

### PR DESCRIPTION
**Pull request by fisherxu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
In v1.7, we add the negotiate part, which negotiate the tunnel port for cloudcore. In this pr, we add the iptables manager, which will add the iptables rules for all the cloudcore automatically.

Now the iptables manager is coupled in cloudcore, in future, we can move it to an independent process when deployment ready.

The iptables rule as below:

```shell
Chain PREROUTING (policy ACCEPT 38 packets, 2248 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  823 92521 DOCKER     all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL
 1633 97916 TUNNEL-PORT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* kubeedge tunnel port */

Chain OUTPUT (policy ACCEPT 91 packets, 6672 bytes)
 pkts bytes target     prot opt in     out     source               destination         
   19  1140 DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL
24329 1505K TUNNEL-PORT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* kubeedge tunnel port */

Chain TUNNEL-PORT (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    1    60 DNAT       tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:10351 to:192.168.1.16:10003
```

**Which issue(s) this PR fixes**:

Ref: https://github.com/kubeedge/kubeedge/issues/1560

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add iptables manager for tunnel port
```

### CODE REVIEW

File: pkg/apis/componentconfig/cloudcore/v1alpha1/validation/validation.go [hunk 1]
```diff
@@ -45,18 +43,6 @@ func ValidateCloudCoreConfiguration(c *v1alpha1.CloudCoreConfig) field.ErrorList
 	return allErrs
 }
 
-//ValidateLeaderElectionConfiguration validates part `l` and returns an errorList if it is invalid, the rest will be validated at run time
-func ValidateLeaderElectionConfiguration(l componentbaseconfig.LeaderElectionConfiguration) field.ErrorList {
```
Comment (chendave) [inline pkg/apis/componentconfig/cloudcore/v1alpha1/validation/validation.go:49]: not pretty sure why this is removed, doesn't that mean `leaderelelction` is no longer supported?

**Review by chendave — COMMENTED** [review]
I guess it's challenging to get those func tested.

/lgtm

File: pkg/apis/componentconfig/cloudcore/v1alpha1/validation/validation.go (hunk 1 above)
Comment (fisherxu) [reply pkg/apis/componentconfig/cloudcore/v1alpha1/validation/validation.go:49]: Now we will support active-active mode of cloudcore, so leaderelelction is no loanger needed. :)

**Comment by chendave** [comment]
let me retest whole CI check.

**Comment by fisherxu** [comment]
> let me retest whole CI check.

The failed e2e test is because the compatibility of kind and github action env, already have an issue to track it. #3002

**Comment by chendave** [comment]
/approve 
/hold
hold in case you want someone else to review this, or else feel free to lift the label.

**Comment by sdghchj** [comment]
mark

**Comment by fisherxu** [comment]
/hold cancel
Let's merge it first, then I will add some e2e test to test this feature. Thanks for the review from @chendave :)

File: cloud/cmd/cloudcore/app/server.go [hunk 2]
```diff
@@ -217,10 +221,10 @@ func NegotiateTunnelPort() (*int, error) {
 
 func negotiatePort(portRecord map[int]bool) int {
 	for port := constants.ServerPort; ; {
+		port++
```
Comment (sdghchj) [inline cloud/cmd/cloudcore/app/server.go:224]: Why start with 10351?

Comment (sdghchj) [inline cloud/cmd/cloudcore/app/server.go:224]: Why it starts with 10351?

**Comment by sdghchj** [comment]
It looks like that rules will be add many times if cloudcore runs in multi-instances mode

**Comment by sdghchj** [comment]
Cloudcore is required to run in host network in order to add rules, isn't it ?

**Comment by kevin-wangzefeng** [comment]
> It looks like that rules will be add many times if cloudcore runs in multi-instances mode

Actually the number of rules will be equal to number of cloudcore instances

**Comment by sdghchj** [comment]
Can it works in k8s deployment?

**Comment by lvfei103650** [comment]
Ask a question, is cloudcore deployed on k8s master node?

**Comment by fisherxu** [comment]
> Ask a question, is cloudcore deployed on k8s master node?

@lvfei103650 Now it's need to run at master node if users need to set iptables through cloudcore, but we need to make it can run on another node.

**Comment by lvfei103650** [comment]
Ask a question, we use nginx for load balancing between cloudcore and edgecore, how can we ensure that port 10000 and port 10004 are loaded to the same cloudcore server?

We use this nginx conf , not useless.
nginx conf：
image

**Pull request merged**